Object.java

```diff
@@ -5,7 +5,12 @@
 import java.io.ObjectInputStream;
 import java.io.ObjectOutputStream;
 
+//The class is called Object because I implement a method of this class
 public class Object {
+
+    //Оur lecturer said to implement only deep cloning(maybe I something not so understood)
+    //If I use this method, the new instance already has a new hash code as well as its fields.
+    // Otherwise, they are identical to the old instance
     public static Object deepClone(Object object){

```

Comment:
Deep clone is override of clone method in your class  and create copies of its fields if they are reference. Here is serialization of object and you do not have destination to where serialize your object  and where read it from.This code will not work. Firstly Object has no fields what is the purpose to serialize it ?